### PR TITLE
travis: Use Ninja for Travis builds

### PR DESCRIPTION
**Pull request by ccawley2011**
Based on yuzu-emu/yuzu#478. Requires citra-emu/build-environments#6.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4125)

### CODE REVIEW

File: .travis/transifex/docker.sh [hunk 1]
```diff
@@ -26,8 +26,8 @@ tx --version
 
 cd /citra
 mkdir build && cd build
-cmake .. -DENABLE_QT_TRANSLATION=ON -DGENERATE_QT_TRANSLATION=ON -DCMAKE_BUILD_TYPE=Release -DENABLE_SDL2=OFF
-make translation
+cmake .. -G Ninja -DENABLE_QT_TRANSLATION=ON -DGENERATE_QT_TRANSLATION=ON -DCMAKE_BUILD_TYPE=Release -DENABLE_SDL2=OFF
```
Comment (jroweboy) [inline .travis/transifex/docker.sh]: i don't really see much point in using ninja for generating qt translations to be quite honest. i'd prefer it if we stuck with make for this build

**Comment by wwylele** [comment]
Is CI failure caused by this PR or it is random error?

**Comment by wwylele** [comment]
Please DO NOT MERGE THIS until #4207 is resolved

**Comment by zhaowenlan1779** [comment]
As is stated in that issue, you may want to add some arguments (`BUILD_BYPRODUCTS`) to the libzmq build to fix #4207

**Comment by wwylele** [comment]
We removed libzmq. Want to revive this?

**Comment by ccawley2011** [comment]
> We removed libzmq. Want to revive this?

I'm happy to, however this PR still requires citra-emu/build-environments#6 to be merged before the Linux builds will work.

(I'm not entirely sure what's wrong with the Mac OS X build, though)

**Comment by zhaowenlan1779** [comment]
That PR is merged. I'll restart the build once docker hub finishes building the images.

**Comment by zhaowenlan1779** [comment]
macos failure looks real

**Comment by ccawley2011** [comment]
> macos failure looks real, can you please look into it?

That's strange, since it failed with a different (possibly unrelated) issue before the build was restarted. I'm not sure what the issue there is either, however I dropped the Mac OS X change to stop it from blocking the PR. If someone is sufficiently interested, it can be revisited later.

**Comment by zhaowenlan1779** [comment]
please resolve conflicts

**Pull request merged**